dali/operators/reader/nemo_asr_reader_op.cc

```diff
@@ -183,18 +185,21 @@ void NemoAsrReader::Prefetch() {
     for (int i = 0; i < nsamples; i++) {
       auto it = decoded_map_.find(curr_batch[i].get());
       if (it != decoded_map_.end() && it->second != i) {
-        audio_batch[i].Copy(audio_batch[it->second]);
+        audio_batch.UnsafeCopySample(i, audio_batch, it->second);
       }
     }
   }
 }
 
 void NemoAsrReader::RunImpl(SampleWorkspace &ws) {
   const auto &sample = GetSample(ws.data_idx());
-  const auto &sample_audio = GetDecodedAudioSample(ws.data_idx());
+  auto sample_audio = GetDecodedAudioSample(ws.data_idx());
 
   auto &audio = ws.Output<CPUBackend>(0);
-  audio.Copy(sample_audio);
+  audio.Resize(sample_audio.shape(), sample_audio.type());
+  std::memcpy(
+      audio.raw_mutable_data(), sample_audio.raw_data(),
+      sample_audio.shape().num_elements() * TypeTable::GetTypeInfo(sample_audio.type()).size());
```

Comment:
Isn't it that GetTypeInfo introduces some locking while we could get this from, let sat GetDecodedAudioSample() for free?

Reply:
Hmm, I can get it via the type info of the `audio` that we just resized.